### DISCUSSION TITLE
[Question] Get new events after bot reconnect?

### DISCUSSION THREAD

**Discussion by kevinvenclovas** [discord-net/Discord.Net#2393] [Q&A] [answered]
Hi,

one question about the event : UserJoined
When a UserJoined my server i want to send him a welcome message. 

But here is my case.

My bot is offline and someone joined my server so he the bot can't send the welcome message.
Is there an queue or something when my bot comes back online to get the events and send all welcome message to the missed events?

**Comment by SylveonDeko** [comment]
Nope, discord doesn't work like that

**Comment by SylveonDeko** [comment]
Most you can do is cache the list of users and check it after the bot starts up

**Comment by csmir** [comment, marked as answer]
As mentioned above, Discord does not send any messages to offline bots, no socket connection is established. An alternative has also been mentioned. You can store the users last inside the server with json or an alternative, fetch the list when the bot starts back up, and compare it to the new list. You can extract the new entries that don't exist in the prior, and send a message to those users. 

Please be careful though; If you do this for a large amount of users, it is considered mass DM, which is against the Discord TOS.
